### PR TITLE
Add Button component

### PR DESCRIPTION
**Pull request by mitchthorson**
Adds basic `<Button />` based on design standards. Includes variants:
- primary
- primary-black
- secondary
- secondary-black
- tertiary

And sizes:
- standard
- small

### CODE REVIEW

File: src/lib/Button/Button.stories.svelte [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          component: "Basic HTML Button adhering to Urban styles."
+        }
+      }
+    }
+  };
+</script>
+
+<script>
+  import { Story, Template } from "@storybook/addon-svelte-csf";
+  import { fireEvent, within, expect } from "@storybook/test";
```
Comment (benkates) [inline src/lib/Button/Button.stories.svelte:34]: I see this referenced but unused, was the intention to test this component?

File: src/lib/Button/Button.svelte [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    // not hovered colors
+    if (_variant === "secondary" || _variant == "secondary-black" || _variant === "tertiary") {
+      return urbanColors.white;
+    }
+    return urbanColors.black;
+  }
+
+  $: iconColor = getIconColor(variant, hovered);
+</script>
+<button class="variant-{variant} size-{size}" on:click {disabled} on:mouseenter={(e) => hovered = true} on:mouseleave={(e) => hovered = false}>
```
Comment (benkates) [inline src/lib/Button/Button.svelte:55]: Should we be adding an event dispatch for click, mouseenter, and mouseleave?

**Review by benkates — COMMENTED** [review]
Looking good, wondering about testing and event dispatching!

File: src/lib/Button/Button.svelte (hunk 2 above)
Comment (mitchthorson) [reply src/lib/Button/Button.svelte:55]: yes, good call on mouseenter and mouseleave

**Comment by mitchthorson** [comment]
Tests added to make sure `click`, `mouseenter` and `mouseleave` events work as expected. Needed to tweak one build setting to allow these tests to run successfully.

**Pull request merged**